File: Tree/104. Maximum Depth of Binary Tree.java
**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int max=0;
    public int maxDepth(TreeNode root) {
        return maxDepth2(root, 0);
    }
    private int maxDepth2(TreeNode root, int level){
        if(root==null){
            return level;
        }
        int left = maxDepth2(root.left,level+1);
        int right = maxDepth2(root.right,level+1);
        int localMax = (left>right) ? left : right;
        max = (localMax>max) ? localMax : max;

        return max;
    }
}

File: Tree/104. Maximum Depth of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int max=1;
    int maxDepth(TreeNode* root) {
        if(root==nullptr){
            return 0;
        }
        Diving(root,1);
        return max;
    }
    void Diving(TreeNode* root,int count){
        if(root->left==nullptr&&root->right==nullptr){
            if(max<count){
                max=count;
            }
            return;
        }
        if(root->left!=nullptr){
            Diving(root->left,count+1);
        }
        if(root->right!=nullptr){
            Diving(root->right,count+1);
        }
    }
};